R/table_parts.R

```diff
@@ -253,32 +256,75 @@ tab_row_group <- function(data,
 #' @export
 tab_spanner <- function(data,
                         label,
-                        columns) {
+                        columns,
+                        gather = TRUE) {
   checkmate::assert_character(
     label, len = 1, any.missing = FALSE, null.ok = FALSE)
 
-  # If using the `vars()` helper, get the columns as a character vector
-  if (inherits(columns, "quosures")) {
-    columns <- columns %>% lapply(`[[`, 2) %>% as.character()
-  }
+  data_df <- as.data.frame(data)
+  colnames <- colnames(data_df)
+
+  columns <- enquo(columns)
+
+  resolved_columns <-
+    resolve_vars(var_expr = columns, var_names = colnames, data_df = data_df)
+
+  # Translate the column indices to column names
+  resolved_columns <- colnames[resolved_columns]
 
   # Filter the vector of column names by the
   # column names actually in `input_df`
-  columns <- columns[which(columns %in% colnames(data))]
+  resolved_columns <- resolved_columns[which(resolved_columns %in% colnames)]
 
-  if (length(columns) == 0) {
+  if (length(resolved_columns) == 0) {
     return(data)
   }
 
   grp_labels <- attr(data, "grp_labels", exact = TRUE)
 
-  for (i in seq(columns)) {
-    grp_labels[[columns[i]]] <- label
+  for (i in seq(resolved_columns)) {
+    grp_labels[[resolved_columns[i]]] <- label
```

Comment:
Please vectorize this for loop away (not apply)

Reply:
`FACE PALM` because: `grp_labels[resolved_columns] <- label` !